src/octoprint/static/js/app/helpers.js

```diff
@@ -1616,3 +1620,67 @@ var deepMerge = function (target, source) {
 
     return target;
 };
+
+var negotiateWebRTC = function (streamUrl) {
+    pc.addTransceiver("video", {direction: "recvonly"});
+    pc.addTransceiver("audio", {direction: "recvonly"});
+    return pc
+        .createOffer()
+        .then(function (offer) {
+            return pc.setLocalDescription(offer);
+        })
+        .then(function () {
+            // Wait for ICE gathering to complete
+            return new Promise(function (resolve) {
+                if (pc.iceGatheringState === "complete") {
+                    resolve();
+                } else {
+                    function checkState() {
+                        if (pc.iceGatheringState === "complete") {
+                            pc.removeEventListener("icegatheringstatechange", checkState);
+                            resolve();
+                        }
+                    }
+                    pc.addEventListener("icegatheringstatechange", checkState);
+                }
+            });
+        })
+        .then(function () {
+            var offer = pc.localDescription;
+            streamUrl = streamUrl.slice("webrtc://".length);
+            return fetch("http://" + streamUrl, {
```

Comment:
OctoPrint generally supports older browsers still, not sure if fetch is generally able to be used or not. @foosel  has never given me a definitive list of what browsers OctoPrint will and won't support...

Most of the rest of the codebase uses jQuery/ajax rather than fetch.

Reply:
Ah good catch, would a  `try`/`catch` around the body of `startWebRTC` be enough do you think?

Reply:
(I'm kinda new at JS, apologies if this question is basic)

Reply:
> @foosel has never given me a definitive list of what browsers OctoPrint will and won't support...

Goal for now: ES5 compatibility. That seems to be the deciding factor for many old tablets out there. So:

https://caniuse.com/?search=es5

Whatever else is supported under these goes. The fetch API sadly doesn't, but it shouldn't be too big of an issue to achieve the same with the jquery `ajax` functions if I'm not mistaken.